internal/entitlement/balanceworker/worker.go

```diff
@@ -65,9 +67,14 @@ type highWatermarkCacheEntry struct {
 	IsDeleted     bool
 }
 
+type connectors struct {
+	entitlement *registry.Entitlement
+	repo        BalanceWorkerRepository
+}
+
 type Worker struct {
 	opts       WorkerOptions
-	connectors *registry.Entitlement
+	connectors connectors
```

Comment:
```suggestion
	entitlement *registry.Entitlement
	repo BalanceWorkerRepository
```

I don't think we need another level of grouping here, WDYT?

Reply:
i think i prefer the word "connectors" as `w.connectors.entitlement` reads better to me than `w.entitlement` simply semantically, but I'm fine with either

Reply:
I prefer to not to have repos in a struct called connectors ;-)